File: Programmers/Level1/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n)
{
    bool *arr = new bool[n + 1]{false};
    int answer = 0;

    for(int i = 2; i <= n; i++)
    {
        for(int j = 2; j*i <= n; j++)
            if(arr[i*j] == false)
                arr[i*j] = true;
    }

    for(int i = 2; i <= n; i++)
    {
        if(arr[i] == false)
            answer++;
    }

    return answer;
}

File: Programmers/Level2/소수 찾기.py
from itertools import permutations
import math

def solution(numbers):
    answer = 0
    nums = []
    
    for i in range(1, len(numbers) + 1):
        temp = set(permutations(list(numbers), i))

        for t in temp:
            t = ''.join(t)
            if t[0] != '0' and t != '1':
                nums.append(int(t))
            
    for n in nums:
        is_prime = True
        for i in range(2, int(math.sqrt(n)) + 1):
            if n % i == 0:
                is_prime = False
                break
        if is_prime:
            answer += 1
    
    return answer